packages/compass-crud/src/components/bulk-update-dialog.tsx

```diff
@@ -93,6 +100,131 @@ const updatePreviewStyles = css({
   gap: spacing[3],
 });
 
+const modalFooterToolbarSpacingStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'space-between',
+});
+
+const modalFooterFormActionsStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: spacing[2],
+});
+
+const modalFooterAdditionalActionsStyles = css({});
+
+const inlineSaveQueryModalStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  padding: spacing[3],
+  gap: spacing[3],
+});
+
+const inlineSaveQueryModalInputStyles = css({
+  width: '315px',
+});
+
+type InlineSaveQueryModalProps = {
+  onSave: (name: string) => void;
+};
+
+const inlineSaveQueryModalContainedElements = ['#inline-save-query-modal *'];
+
+const InlineSaveQueryModal: React.FunctionComponent<
+  InlineSaveQueryModalProps
+> = ({ onSave }) => {
+  const [open, setOpen] = useState(false);
+  const [favoriteName, setFavoriteName] = useState('');
+  const [valid, setValid] = useState(false);
+
+  const cleanClose = useCallback(() => {
+    setOpen(false);
+    setFavoriteName('');
+  }, [setOpen, setFavoriteName]);
+
+  const onClickSave = useCallback(() => {
+    onSave(favoriteName);
+    cleanClose();
+  }, [onSave, favoriteName, cleanClose]);
+
+  const updateFavoriteName = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      const favoriteName: string = ev.target.value || '';
+
+      setFavoriteName(favoriteName);
+      setValid(favoriteName !== '');
+    },
+    [setFavoriteName, setValid]
+  );
+
+  const handleSpecialKeyboardEvents = useCallback(
+    (ev: React.KeyboardEvent) => {
+      if (ev.key === 'Enter') {
+        onSave(favoriteName);
+        cleanClose();
+      } else if (ev.key === 'Escape') {
+        cleanClose();
+      }
+    },
+    [favoriteName, onSave, cleanClose]
+  );
+
+  return (
+    <InteractivePopover
+      containedElements={inlineSaveQueryModalContainedElements}
+      // To prevent popover from closing when confirmation modal is shown
```

Comment:
Is this comment supposed to be for the line above?

Reply:
Yep, but I'm going to remove it, I don't think this comment adds much information. Ty ❤️ .